src/net/download.rs

```diff
@@ -0,0 +1,1904 @@
+                                            Ok(handle) => {
+                                                debug!("{:?}: Begin HTTP request for anchored block {} to {:?} ({:?})", &network.local_peer, &key.index_block_hash, &key.neighbor, &key.data_url);
+                                                requests.insert(key.clone(), handle);
+                                                sent = true;
+                                                break;
+                                            }
+                                            Err(e) => {
+                                                debug!("{:?}: Failed to connect or send HTTP request to {:?} ({:?}, {:?}): {:?}", &network.local_peer, &key.neighbor, &key.data_url, addr, &e);
+                                                continue;
+                                            }
+                                        }
+                                    }
+
+                                    if !sent {
+                                        debug!("{:?}: Failed to request block {:?} from {:?}", &network.local_peer, &key.index_block_hash, sockaddrs);
+                                        continue;
+                                    }
+                                }
+                                else {
+                                    debug!("{:?}: Will not fetch block {:?}: failed to look up DNS name in {:?}", &network.local_peer, &key.index_block_hash, &key.data_url);
+                                    continue;
+                                }
+                            },
+                            None => {
+                                debug!("{:?}: No more requests for block at sortition height {}", &network.local_peer, sortition_height);
+                                continue;
+                            }
+                        }
+                    },
+                    None => {
+                        debug!("{:?}: No block at sortition height {}", &network.local_peer, sortition_height);
+                    }
+                }
+            }
+
+            downloader.getblocks_begin(requests);
+            Ok(())
+        })
+    }
+
+    /// Try to see if all blocks are finished downloading
+    pub fn block_getblocks_try_finish(&mut self) -> Result<bool, net_error> {
+        test_debug!("{:?}: block_getblocks_try_finish", &self.local_peer);
+        PeerNetwork::with_downloader_state(self, |ref mut network, ref mut downloader| {
+            match network.http {
+                Some(ref mut http) => {
+                    downloader.getblocks_try_finish(http)
+                },
+                None => {
+                    test_debug!("{:?}: HTTP not connected", &network.local_peer);
+                    Err(net_error::NotConnected)
+                }
+            }
+        })
+    }
+
+    /// Proceed to get microblocks 
+    pub fn block_getmicroblocks_begin(&mut self) -> Result<(), net_error> {
```

Comment:
This and `block_getblocks_begin` could probably be deduped.

Reply:
Will do.

Reply:
Fixed in 2fccb98